### PR TITLE
Add SQLAlchemy to python libraries

### LINKED ISSUE #1745: document using sqlalchemy to access RisingWave
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

Python users may be more familar with sqlalchemy than [psycopg2](https://pypi.org/project/psycopg2/)

### Reference

_No response_

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  - Add SQLAlchemy to python libraries

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/14491

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1745

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/dev/python-client-libraries.md [hunk 1]
```diff
@@ -103,4 +104,54 @@ with conn.cursor() as cur:
...
+### Create a source
+
+The code below creates a table `users` using the engine created in the previous section.
+
+```python
+with engine.connect() as conn:
+    conn.execute("""CREATE TABLE IF NOT EXISTS users (
+    id INTEGER PRIMARY KEY,
+    name VARCHAR,
+    age INTEGER)""")
```
Comment (ShanlanLi) [inline docs/dev/python-client-libraries.md:150]: Does it need a ";"?

Comment (ShanlanLi) [reply docs/dev/python-client-libraries.md:150]: I see some codes in this topic have ";", while some don't, including:

```
...
with conn.cursor() as cur:
    cur.execute("""
CREATE TABLE walk(distance INT, duration INT)
WITH (
    connector = 'datagen',
    fields.distance.kind = 'sequence',
    fields.distance.start = '1',
    fields.distance.end  = '60',
    fields.duration.kind = 'sequence',
    fields.duration.start = '1',
    fields.duration.end = '30',
    datagen.rows.per.second='15',
    datagen.split.num = '1'
) FORMAT PLAIN ENCODE JSON""") # Execute the query.
...
```
I'm not sure if it needs ";". Please check it.

**Review by ShanlanLi — APPROVED** [review]
Rest LGTM, thanks.

File: docs/dev/python-client-libraries.md (hunk 1 above)
Comment (emile-00) [reply docs/dev/python-client-libraries.md:150]: No need for `;` based off the integration tests. Might be different requirements for the type of query? @neverchanje Can you help double check this?

Comment (neverchanje) [reply docs/dev/python-client-libraries.md:150]: Both `;` and none are all acceptable to our SQL parser. That's affirmative.

File: docs/dev/python-client-libraries.md [hunk 2]
```diff
@@ -103,4 +104,54 @@ with conn.cursor() as cur:
...
+```
+
+You can create materialized views and query from materialized views using the same format shown above. 
+
+:::note
+When creating tables using SQLAlchemy, `BIGINT` types will automatically be converted to `BIGSERIAL` types, which is not supported in RisingWave.
```
Comment (neverchanje) [inline docs/dev/python-client-libraries.md]:
```suggestion
When creating tables using SQLAlchemy 2.0, `BIGINT` types will automatically be converted to `BIGSERIAL` types, which is not supported in RisingWave.
```

**Review by neverchanje — APPROVED** [review]
Thanks. The refined doc looks much better and the content is correct.

**Pull request merged**